packages/issuer-api/src/pods/certification-request/handlers/approve-certification-request.handler.ts

```diff
@@ -63,16 +64,18 @@ export class ApproveCertificationRequestHandler
             return ResponseFailure(e.message, HttpStatus.INTERNAL_SERVER_ERROR);
         }
 
-        this.eventBus.publish(
-            new CertificateCreatedEvent(newCertificateId, isPrivate ? { owner, energy } : null)
-        );
-
         await this.repository.update(id, {
             approved: true,
             approvedDate: new Date(),
             issuedCertificateTokenId: newCertificateId
         });
 
+        this.eventBus.publish(
+            new CertificateCreatedEvent(newCertificateId, isPrivate ? { owner, energy } : null)
+        );
+
+        this.eventBus.publish(new CertificateRequestApprovedEvent(id));
```

Comment:
What's the purpose of the `CertificateRequestApprovedEvent` ?

Reply:
CertificateCreated event fired too often and there is no way I can check if I already sent an email. So instead of I decided to create a separate event which I can fully control.